frontend/src/components/input/ListInput.vue

```diff
@@ -0,0 +1,145 @@
+<script lang="ts" setup>
+import { ref, computed, watch, nextTick, onMounted } from "vue"
+import Checkbox from "@/components/input/CheckboxInput.vue"
+import InputField, { LabelPosition } from "@/components/input/InputField.vue"
+import TextButton from "@/components/input/TextButton.vue"
+
+const props = defineProps<{
+  label: string
+  modelValue: string[]
+}>()
+
+const emit = defineEmits<{
+  "update:modelValue": [value: string[]]
+}>()
+
+const editMode = ref(true)
+const sortAlphabetically = ref(false)
+const localModelValue = ref(props.modelValue ?? [])
+
+const textAreaInput = computed({
+  get: () => (props.modelValue ? props.modelValue.join("\n") : ""), // Join array with newlines for textarea
+  set: (newValues: string) => {
+    // Split the text by newline, trim each line, filter out empty lines, and set back into the array
+    localModelValue.value = newValues
+      .split("\n")
+      .map((listitem) => listitem.trim())
+      .filter((listitem) => listitem !== "")
+  },
+})
+
+const addList = () => {
+  // sort alphabetically if option set
+  if (sortAlphabetically.value && localModelValue.value) {
+    localModelValue.value = localModelValue.value.sort((a: string, b: string) =>
+      a.localeCompare(b),
+    )
+  }
+  emit("update:modelValue", [...new Set(localModelValue.value)] as string[]) //remove duplicates
+  if (!!localModelValue.value?.length) editMode.value = false
+  sortAlphabetically.value = false
+}
+const cancelEdit = () => {
+  if (!!localModelValue.value?.length) editMode.value = false
```

Comment:
For me as a user, it was confusing that I cannot leave the edit mode when there is no input. I thought it was not saving it. I asked Agnieszka [here](https://www.figma.com/design/oyeWgl6BRJA15tkGjUD6Ze?node-id=18303-63930#962540636).

Reply:
Agnieszka wants the state to go back to the CategoryWrapper button. Maybe we can add an optional reset handler to the CategoryWrapper that we pass to the slot, see https://vuejs.org/guide/components/slots.html#scoped-slots

Reply:
Thank you for evaluating with Agnieszka, added it here :)
https://github.com/digitalservicebund/ris-backend-service/pull/2187/commits/8527415da81ffc265afa471593395bd37ae083c1